ably/params.go

```diff
@@ -0,0 +1,137 @@
+package ably
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+)
+
+type ProtocolType string
+
+const (
+	ProtocolJSON    = "json"
+	ProtocolMsgPack = "msgpack"
+)
+
+type Params struct {
+	RealtimeEndpoint string
+	RestEndpoint     string
+
+	ApiKey       string
```

Comment:
These comments are valid but they all stem from the fact that it is code that was already in that state before the refactor.

@rjeczalik You should address them in another PR. Let's focus on getting this package convention out first.